compiler/src/iree/compiler/Codegen/LLVMGPU/ConvertToROCDL.cpp

```diff
@@ -120,6 +143,8 @@ struct ConvertToROCDLPass : public ConvertToROCDLBase<ConvertToROCDLPass> {
       populateFuncToLLVMConversionPatterns(converter, llvmPatterns);
       cf::populateControlFlowToLLVMConversionPatterns(converter, llvmPatterns);
       arith::populateArithToLLVMConversionPatterns(converter, llvmPatterns);
+      FailureOr<amdgpu::Chipset> maybeChipset = amdgpu::Chipset::parse("gfx90a");
```

Comment:
I think this will break the flow for other architectures on this path? I think this problem is solved on the cuda side over here
https://github.com/openxla/iree/blob/7a50b093af95b0bb0fad8ded530827aa2dbbdaa5/compiler/src/iree/compiler/Codegen/LLVMGPU/ConvertToNVVM.cpp#L42-L65
This does seem out of scope for this PR though and can come later, so I would recommend that you guard this with a flag for the time being. just a boolean flag on if we want to use these passes will do?
Here is an example of a boolean flag
https://github.com/openxla/iree/blob/7a50b093af95b0bb0fad8ded530827aa2dbbdaa5/compiler/src/iree/compiler/Dialect/HAL/Target/ROCM/ROCMTarget.cpp#L39-L42

Reply:
If you do want to take on the correct way of doing how cuda does it then I would look at kernelconfig.cpp on the parallels between the archs in cuda and rocm to make your own `getRocmChip` function like the `getCUDACapbility`

Reply:
Just updated.